packages/react/slider/src/Slider.tsx

```diff
@@ -486,28 +488,44 @@ SliderRange.displayName = RANGE_NAME;
 const THUMB_NAME = 'SliderThumb';
 const THUMB_DEFAULT_TAG = 'span';
 
-type SliderThumbOwnProps = Omit<Polymorphic.OwnProps<typeof SliderThumbImpl>, 'value' | 'index'>;
+type SliderThumbOwnProps = Omit<Polymorphic.OwnProps<typeof SliderThumbImpl>, 'index'>;
 type SliderThumbPrimitive = Polymorphic.ForwardRefComponent<
   Polymorphic.IntrinsicElement<typeof SliderThumbImpl>,
   SliderThumbOwnProps
 >;
 
 const SliderThumb = React.forwardRef((props, forwardedRef) => {
-  const { ref: collectionRef, index } = useSliderCollectionItem();
-  const ref = useComposedRefs(forwardedRef, collectionRef);
-  const context = useSliderContext(THUMB_NAME);
-  const value = context.values[index];
-  return value !== undefined ? (
-    <SliderThumbImpl {...props} ref={ref} index={index} value={value} />
+  const { getItems } = useCollection();
+  const [thumb, setThumb] = React.useState<React.ElementRef<typeof SliderThumbImpl> | null>(null);
+  const composedRefs = useComposedRefs(forwardedRef, (node) => setThumb(node));
+  const isInitialRenderRef = React.useRef(true);
+  const index = React.useMemo(
+    () => (thumb ? getItems().findIndex((item) => item.ref.current === thumb) : -1),
+    [getItems, thumb]
+  );
+
+  React.useEffect(() => {
+    isInitialRenderRef.current = false;
+  }, []);
+
+  /**
+   * Until we figure out SSR indexes, we hide the thumbs on initial render while we work out
+   * the index. Otherwise, SSR will place the thumbs in the wrong position and snap into the
+   * right position during hydration which would be jarring.
+   */
```

Comment:
```suggestion
 /**
   * We hide thumbs on the first render otherwise SSR will initially place them
   * in the wrong position and later snap them into the correct position during hydration
   * which would be visually jarring.
   * 
   * Previously we accounted for this by managing indexes inside the collection ahead of time,
   * but this caused problems with `React.StrictMode` so we're falling back to this for now.
   */
```

Nit: How about something like this?